plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextJavaValidator.java

```diff
@@ -560,65 +617,66 @@ public void checkAnnotations(final Statechart statechart) {
+		Map<Region, List<Entry>> regions = null;
+
+		// first list contains Transitions without entry spec
+		if (!transitions[0].isEmpty()) {
+			regions = STextValidationModelUtils.getRegionsWithoutDefaultEntry(state.getRegions());
+			if (!regions.isEmpty()) {
+				for (Transition transition : transitions[0]) {
+					error(TRANSITION_UNBOUND_DEFAULT_ENTRY_POINT, transition, null, -1);
+				}
+				for (Region region : regions.keySet()) {
+					error(REGION_UNBOUND_DEFAULT_ENTRY_POINT, region, null, -1);
 				}
 			}
+		}
 
-			// second list contains Transitions with entry spec
-			if (!transitions[1].isEmpty()) {
-				if (regions == null) {
-					regions = STextValidationModelUtils.getRegionsWithoutDefaultEntry(state.getRegions());
-				}
-				for (Transition transition : transitions[1]) {
-					boolean hasTargetEntry = true;
-					for (ReactionProperty property : transition.getProperties()) {
-						if (property instanceof EntryPointSpec) {
-							EntryPointSpec spec = (EntryPointSpec) property;
-							String specName = "'" + spec.getEntrypoint() + "'";
-							for (Region region : regions.keySet()) {
-								boolean hasEntry = false;
-								for (Entry entry : regions.get(region)) {
-									if (entry.getName().equals(spec.getEntrypoint())) {
-										hasEntry = true;
-										break;
-									}
-								}
-								if (!hasEntry) {
-									error(REGION_UNBOUND_NAMED_ENTRY_POINT + specName, region, null, -1);
-									hasTargetEntry = false;
+		// second list contains Transitions with entry spec
+		if (!transitions[1].isEmpty()) {
+			if (regions == null) {
+				regions = STextValidationModelUtils.getRegionsWithoutDefaultEntry(state.getRegions());
+			}
+			for (Transition transition : transitions[1]) {
+				boolean hasTargetEntry = true;
+				for (ReactionProperty property : transition.getProperties()) {
+					if (property instanceof EntryPointSpec) {
+						EntryPointSpec spec = (EntryPointSpec) property;
+						String specName = "'" + spec.getEntrypoint() + "'";
+						for (Region region : regions.keySet()) {
+							boolean hasEntry = false;
+							for (Entry entry : regions.get(region)) {
+								if (entry.getName().equals(spec.getEntrypoint())) {
+									hasEntry = true;
+									break;
 								}
 							}
-							if (!hasTargetEntry) {
-								error(TRANSITION_UNBOUND_NAMED_ENTRY_POINT + specName, transition, null, -1);
+							if (!hasEntry) {
+								error(REGION_UNBOUND_NAMED_ENTRY_POINT + specName, region, null, -1);
+								hasTargetEntry = false;
 							}
-							boolean usingEntry = false;
-							for (Region region : state.getRegions()) {
-								EList<Vertex> vertices = region.getVertices();
-								for (Vertex vertice : vertices) {
-									if (vertice instanceof Entry) {
-										if (spec.getEntrypoint().equals(vertice.getName())) {
-											usingEntry = true;
-										}
+						}
+						if (!hasTargetEntry) {
+							error(TRANSITION_UNBOUND_NAMED_ENTRY_POINT + specName, transition, null, -1);
```

Comment:
I have a feeling that you do this way too often. Should go to the outer for loop, looping through the transitions.